src/cryptosettings.cpp

```diff
@@ -12,16 +12,66 @@
 #include "hacl-star/Hacl_Chacha20Poly1305_32.h"
 #include "logger.h"
 
+#if defined(UNIT_TEST)
+#  include "platforms/dummy/dummycryptosettings.h"
+#elif defined(MZ_FLATPAK)
+#  include "platforms/linux/xdgcryptosettings.h"
+#elif defined(MZ_LINUX)
+#  include "platforms/linux/linuxcryptosettings.h"
+#elif defined(MZ_MACOS) || defined(MZ_IOS)
+#  include "platforms/macos/macoscryptosettings.h"
+#elif defined(MZ_WINDOWS)
+#  include "platforms/windows/windowscryptosettings.h"
+#elif defined(MZ_ANDROID)
+#  include "platforms/android/androidcryptosettings.h"
+#endif
+
 constexpr int NONCE_SIZE = 12;
 constexpr int MAC_SIZE = 16;
+constexpr int ENCRYPTED_V2_HEADER_SIZE = 4;
 
 namespace {
-
 Logger logger("CryptoSettings");
+CryptoSettings* s_instance = nullptr;
+}  // namespace
 
-uint64_t lastNonce = 0;
+// static
+QSettings::Format CryptoSettings::format() {
```

Comment:
I think this function does 2 things: 
-> be a factory for cryptosettings 
-> now returning a QSettingsFormat. 

i get why format is creating it (after reading the cpp ) but just by reading the header it's not clear why it would do that. Consider splitting the factory and format creator
```
if (!s_instance) {
 CryptoSettings::create()
}
```

Reply:
I will split this into a separate `CryptoSettings::create()` method.